lib/simplifyExpression/substituteScopeSearch/index.js

```diff
@@ -0,0 +1,60 @@
+const ChangeTypes = require('../../ChangeTypes');
+const Node = require('../../node');
+const TreeSearch = require('../../TreeSearch');
+
+// Searches through the tree, prioritizing deeper nodes, and substitutes
+// in-scope values for their respective expressions on a symbol node if possible.
+// Returns a Node.Status object.
+const search = TreeSearch.postOrder(scopeSubstitution);
+
+function scopeSubstitution(node, options={}) {
+  if (Node.Type.isSymbol(node)) {
+    return substituteAndSimplifySymbols(node, options);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// SUBSTITUTES
+// Returns a Node.Status object with substeps
+function substituteAndSimplifySymbols(node, options={}) {
+  if (!options.hasOwnProperty('scope')) {
+    return Node.Status.noChange(node);
+  }
+
+  if (!Node.Type.isSymbol(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  let symbolName;
+  if (node.type === 'SymbolNode') {
+    symbolName = node.name;
+  }
+  else if (Node.Type.isUnaryMinus(node)) {
```

Comment:
now that I updated the `Node.Type.isSymbol` definition, you'll never get unary minuses in here anymore which makes things much easier hopefully!

so you can take out `if (node.type === 'SymbolNode') {` and `else if (Node.Type.isUnaryMinus(node)) {` and always just do ` symbolName = node.name;`